service/api_plugin_git.go

```diff
@@ -4,27 +4,67 @@ import (
 	"fmt"
 	"net/http"
 
+	"github.com/evergreen-ci/evergreen"
 	"github.com/evergreen-ci/evergreen/db"
+	"github.com/evergreen-ci/evergreen/model/build"
 	"github.com/evergreen-ci/evergreen/model/patch"
+	"github.com/evergreen-ci/evergreen/model/task"
 	"github.com/evergreen-ci/gimlet"
 	"github.com/pkg/errors"
 )
 
 func (as *APIServer) gitServePatch(w http.ResponseWriter, r *http.Request) {
-	task := MustHaveTask(r)
+	t := MustHaveTask(r)
 
-	patch, err := patch.FindOne(patch.ByVersion(task.Version))
+	p, err := patch.FindOne(patch.ByVersion(t.Version))
 	if err != nil {
 		as.LoggedError(w, r, http.StatusInternalServerError,
-			errors.Wrapf(err, "problem fetching patch for task '%s' from db", task.Id))
+			errors.Wrapf(err, "problem fetching patch for task '%s' from db", t.Id))
 		return
 	}
-	if patch == nil {
+	if p == nil {
 		as.LoggedError(w, r, http.StatusNotFound,
-			errors.Errorf("no patch found for task %s", task.Id))
+			errors.Errorf("no patch found for task %s", t.Id))
 		return
 	}
-	gimlet.WriteJSON(w, patch)
+
+	// add on the merge status for the patch, if applicable
+	if p.GetRequester() == evergreen.MergeTestRequester {
+		builds, err := build.Find(build.ByVersion(p.Version))
+		if err != nil {
+			as.LoggedError(w, r, http.StatusInternalServerError, errors.Wrap(err, "error retrieving builds for task"))
+			return
+		}
+		tasks, err := task.Find(task.ByVersion(p.Version).WithFields(task.BuildIdKey, task.StatusKey, task.ActivatedKey, task.DependsOnKey, task.IsGithubCheckKey))
```

Comment:
Adding WithFields makes it pretty easy to create future bugs that are hard to find; considering how many fields already you're keeping, it may not be worth introducing that risk, considering this part of the function isn't called all the time or anything.

Reply:
If you do choose to keep WithFields, GithubChecks isn't necessary (I see where you copied this from, but it's relevant to mainline commits only)

Reply:
removed the github check. I think that projecting a couple fields reduces the response size enough that it's worth doing